### ISSUE TITLE
Unable to run multiple feature files using multiple tags

### ISSUE THREAD

**Issue by TabeicaRemus** [mucsi96/nightwatch-api#832]
Hi! I am facing a problem and I don't know how to find a solution for it since I've tried a couple of scenarios but without any success yet. I have 2 feature files, the first feature called login.feature has the **@login tag** for each scenario and the other feature file is called users.feature with **@users tag** under each scenario.

## Expected Behavior
Assuming you have something like this in package.json:
`"test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json --tags=@login,@users"`

Running script in this way: npm run test

I am expecting to see that first of all, are triggered to run the tests from login.feature and after that to run the tests from users.feature file but instead of happening this, I am getting this output:

**_Output:_**
**0 scenarios
0 steps
0m00.000s**

## Current Behavior
If I am adding the tags in this way: --tags=@login,@users, I see this output in the IntelliJ's console:

**0 scenarios
0 steps
0m00.000s**
At this point, no scenario is triggered to run, only the browser opens and closes right away without performing any action on the site.

**OR**

If I am adding the tags in this way: --tags @login and @users, I see this output in the IntelliJ's console:

**Error: ENOENT: no such file or directory, open 'D:\Project_Directory\@users'**

// FYI, with a single tag, --tags @login, it works fine!

## Possible Solution

Not sure.

## Steps to Reproduce (for bugs)

_These are my feature files:_
**login.feature**

Feature: Login on my site

  @login
  Scenario: Users can login
    Given I am on the login page
    And I input the "john.doe@test.com" and "pass" credentials
    When I click on the login button
    Then I am redirected to the private area

**users.feature**
Feature: Users - private area

  @users
  Scenario: Functionality on users page
    Given I am on the users page
    And I search for "John Doe" user
     Then I click on delete entry
  And I verify that user section is empty

**This is my package.json**

`{
  "name": "nightwatch-api-cucumber",
  "version": "1.0.0",
  "description": "Demonstration for using Nightwatch API from CucumberJS test runner",
  "main": "index.js",
  "scripts": {
    "test:chrome": "cross-env NIGHTWATCH_ENV=chrome npm test",
    "test:firefox": "cross-env NIGHTWATCH_ENV=firefox npm test",
    "test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json --tags=@login,@users",
    "debug": "cross-env DEBUG=nightwatch-api NIGHTWATCH_ENV=chrome NIGHTWATCH_VERBOSE=true node --inspect ./node_modules/cucumber/bin/cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty"
  },
  "repository": "git@github.com:mucsi96/nightwatch-api.git",
  "author": "Igor Mucsicska <mucsi_96@yahoo.com>",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.8.3",
    "@babel/preset-env": "^7.8.3",
    "@babel/register": "^7.8.3",
    "chromedriver": "^79.0.0",
    "cross-env": "^6.0.3",
    "cucumber": "^6.0.5",
    "cucumber-html-reporter": "^5.0.0",
    "cucumber-pretty": "^6.0.0",
    "geckodriver": "^1.19.1",
    "mkdirp": "^0.5.1",
    "nightwatch": "^1.3.2",
    "nightwatch-api": "3.0.1",
    "npm-run-all": "^4.1.5",
    "moment": "^2.24.0"
  }
}`

And these are the test cases that I've already tried by switching to different approaches when using tags. You will find here also the console's output for every case:

**1. --tags=@login,@users**
Output:0 scenarios
0 steps
0m00.000s

**2. --tags=@login --tags=@users**
Output:0 scenarios
0 steps
0m00.000s

**3. --tags @login,@users**
Output:0 scenarios
0 steps
0m00.000s

**4. --tags @login and @users**
Error: ENOENT: no such file or directory, open 'D:\Project_Directory\@users'

**5. --tags '@login,@users'**
Output:0 scenarios
0 steps
0m00.000s

**6. --tags '@login and @users'**
Error: ENOENT: no such file or directory, open 'D:\Project_Directory\@users''

## Context

See above.

## Your Environment

- Version used: Node v10.16.0, nightwatch-api: 3.0.1
- Browser Name and version: chrome (80.0.3987.106)
- Operating System and version (desktop or mobile): Windows 10 Pro

**Comment by tejasshekokar** [comment]
+1 I'm also facing the same issue.

**Comment by mucsi96** [comment]
Thank you for your feedback. I need some additional information to be able to reproduce the issue. How are you providing the tags? What command are you executing in terminal? Or are you putting these tags in package json? Even your issue in not related to this package as it is not providing any runner I am happy to help.

**Comment by mucsi96** [comment]
My guess is you would face the same issue with pure cucumberjs without even using this package

**Comment by mucsi96** [comment]
So I can try to help with this issue. But I fail you should check with CucumberJS community

**Comment by TabeicaRemus** [comment]
Hi Igor,

**These are my tag scenarios I tried so far for testing with multiple tags.**

1. --tags=@login,@users
2. --tags=@login--tags=@users
3. --tags @login,@users
4. --tags @login and @users
5. --tags '@login,@users'
6. --tags '@login and @users'

**I just replaced the use case in test script attached bellow:**

"test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json **<tagScenario>**"

Here is some example that aren't working:

=> "test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json **--tags=@login,@users**"

=>  "test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json **--tags=@login --tags=@users**"


Just replace **<tagScenario>**  with every tag scenarios from above (the ones from 1 to 6) and you will get this output:

**0 scenarios
0 steps
0m00.000s**

OR

**Error: ENOENT: no such file or directory, open 'D:\Project_Directory@users''**

**This is the command I use to run in IntelliJ's terminal:**
npm run test

**Comment by mucsi96** [comment]
Have you read this section? https://nightwatch-api.netlify.com/cucumber/#feature-and-scenario-tags 
I think the problem will be with single quotes. Try double.

**Comment by TabeicaRemus** [comment]
![Unexpected token @ in JSON](https://user-images.githubusercontent.com/31964285/74679080-7fa2bc00-51c5-11ea-81c3-215216ee0aec.png)

By adding double quotes, I am getting an error: "JSON standard does not allow such tokens". It seems that relates to the fact that I already have the script place in double quotes and you can't place double quotes expression inside of another double quotes.

**_So, it doesn't work with this approach_**: "test": "mkdirp report && cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json --tags "@login or @caregiver""

_By adding the tags inside of a single quote, I am not getting the JSON unexpected token but still, I won't be able to run the tests..._

**Comment by tejasshekokar** [comment]
Hi Igor,

Used double quotes `npm run test:chrome -- --tags "@Editor"`, but it is still throwing the same ERROR.
```
> cucumber-ui-tests@1.0.0 test:chrome web/test
> cross-env NIGHTWATCH_ENV=chrome npm test "--tags" "@Editor"


> cucumber-ui-tests@1.0.0 test web/test
> export TIMESTAMP=$(date +%s) && mkdirp report cucumber && cucumber-js cucumber-js $(cat config/featureOrder.txt) --require-module babel-core/register --require cucumber.conf.js --require step-definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report_$(echo $TIMESTAMP).json "@Editor"

Error: ENOENT: no such file or directory, open 'web/test/@Editor'
npm ERR! Test failed.  See above for more details.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

**Comment by mucsi96** [comment]
If you put the `--tags "@Editor"` part in npm script you have to escape the double quotes using `\"`
I you want to run you test script with adding `-- --tags "@Editor"` you have to make sure that the npm script contain only a single command. This is not true in you case as it runs
- `mkdirp report`
- `cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json`

If you run more then one command in a script you are not able to pass through the arguments using `--` operator. As the command line doesn't know to which command you want to pass it through to `mkdirp` or to `cucumber-js`. Because of this I recommend removing the `mkdirp` from the npm script and add it to `cucumber.conf.js` using the following snippet 

```
try {
  fs.mkdirSync(path.resolve(__dirname, 'report'));
} catch {}
```

Now you can pass through any command line argument to `cucumber-js` using `--` operator. So `npm run test -- --tags "@Editor"` should work.

**Comment by TabeicaRemus** [comment]
Hi Igor,

Thank you for helping us on this matter.

I have some information that it may be useful for others too.

1. By playing with your solution, I've managed to make it work by running the script in this way: `npm run test -- --tags "@login or @users"` _this one works perfectly_, and ran all the scenarios that had login AND users tags.
**_**Info, at this point your tests will run with Headless Chrome option, and the UI is not visible._**

2. For that case where you want to see also the UI while your tests are running, I've managed to make it work by adding tags on my script using also the escape character \"

`"test": "cucumber-js --require-module @babel/register --require cucumber.conf.js --require features/step_definitions --format node_modules/cucumber-pretty --format json:report/cucumber_report.json --tags \"@login or @users or @admin"`

And run the tests using this command: `npm run test:chrome`

![It works for multiple tags](https://user-images.githubusercontent.com/31964285/74733232-93e1ca00-5254-11ea-929c-6bc1addf0671.png)
